pkg/querier/querier.go

```diff
@@ -523,3 +536,29 @@ func (q *Querier) checkTailRequestLimit(ctx context.Context) error {
 
 	return nil
 }
+
+func (q *Querier) queryHasEqualityMatchers(req logql.QueryParams) (bool, error) {
```

Comment:
I'm wondering if it is worth including this function into the logql package. Because I think we need to implement this on the frontend too. It would be great if we could stop this queries at both level (frontend-querier).

@slim-bean should we allow a flag  or a tenant limit to disable ? I'm not a strong fan of having an option but making sure you're onboard ? This is breaking change do we plan this for 2.2 or 2.3 ?

Reply:
@cyriltovena Thanks for the review! I have moved the code to the logql package. I hope I have added it to the right place. Can you please review it and see if it makes sense now?